File: test/benchmark/benchmark.py
import timeit

import wcwidth


PLANE0 = 0xFFFF
PLANE16 = 0x10FFFF


def main():
    started_at = timeit.default_timer()

    for codepoint in range(0, PLANE16 + 1):
        wcwidth.wcwidth(chr(codepoint))

    elapsed = timeit.default_timer() - started_at
    print(elapsed)


if __name__ == "__main__":
    main()

File: test/benchmark/benchmark.exs
defmodule Benchmark do
  def run do
    # Range.new(0x0, 0xFFFF)
    codepoints =
      Range.new(0x0, 0x10FFFF)
      |> Stream.filter(&Advance.UCD.UnicodeData.Parse.valid_codepoint?/1)

    {time, _} =
      :timer.tc(fn ->
        Enum.each(codepoints, fn codepoint -> Advance.of(<<codepoint::utf8>>) end)
      end)

    IO.inspect(time / 1_000_000)
    IO.inspect(:erlang.memory())
  end
end

Benchmark.run()
